samza-kv/src/main/java/org/apache/samza/storage/kv/LargeMessageSafeStore.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.storage.kv;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.samza.SamzaException;
+
+
+public class LargeMessageSafeStore implements KeyValueStore<byte[], byte[]> {
+
+  private KeyValueStore<byte[], byte[]> store;
+  private String storeName;
+  private Boolean dropLargeMessage;
+  private int maxMessageSize;
+
+  public LargeMessageSafeStore(KeyValueStore<byte[], byte[]> store, String storeName, Boolean dropLargeMessage, int maxMessageSize) {
+    this.store = store;
+    this.storeName = storeName;
+    this.dropLargeMessage = dropLargeMessage;
+    this.maxMessageSize = maxMessageSize;
+  }
+
+  @Override
+  public byte[] get(byte[] key) {
+    return store.get(key);
+  }
+
+  @Override
+  public void put(byte[] key, byte[] value) {
+    validateMessageSize(value);
+    if (!isLargeMessage(value)) {
```

Comment:
Need a metric for ignored large messages rate. Should also log them, not ignore them silently.

Reply:
Added a metric and logs.